polaris/hub/polarisfs.py

```diff
@@ -0,0 +1,104 @@
+import fsspec
+
+from typing import Union, List, Dict, Any
+
+from polaris.utils.errors import PolarisHubError
+
+from httpx._types import TimeoutTypes
+
+
+class PolarisFSFileSystem(fsspec.AbstractFileSystem):
+    """
+    A file system interface for accessing datasets on the Polaris platform.
+
+    This class extends `fsspec.AbstractFileSystem` and provides methods to list objects within a Polaris dataset
+    and fetch the content of a file from the dataset.
+
+    Note: Zarr Integration
+        This file system can be used with Zarr for working with multidimensional array data stored in the Polaris dataset.
+
+    Args:
+        polaris_client: The Polaris Hub client used to make API requests.
+        dataset_owner: The owner of the dataset.
+        dataset_name: The name of the dataset.
+        default_expirations_seconds: Default expiration time for signed URLs.
+        **kwargs: Additional keyword arguments.
+    """
+
+    sep = "/"
+    protocol = "polarisfs"
+    async_impl = False
+
+    def __init__(
+        self,
+        polaris_client: Any,
+        dataset_owner: str,
+        dataset_name: str,
+        default_expirations_seconds: int = 10 * 60,
+        **kwargs: dict,
+    ):
+        super().__init__(**kwargs)
+
+        self.polaris_client = polaris_client
+
+        # Prefix to remove from ls entries
+        self.prefix = f"dataset/{dataset_owner}/{dataset_name}/"
+        self.base_path = f"/storage/{self.prefix}"
+
+    def ls(
+        self, path: str, detail: bool = False, timeout: TimeoutTypes = (10, 200), **kwargs: dict
+    ) -> Union[List[str], List[Dict[str, Any]]]:
+        """List objects in the specified path within the Polaris dataset.
+
+        Args:
+            path: The path within the dataset to list objects.
+            detail: If True, returns detailed information about each object.
+            **kwargs: Additional keyword arguments.
+
+        Returns:
+            A list of dictionaries if detail is True; otherwise, a list of object names.
+        """
+        ls_path = f"{self.base_path}ls/{path}"
+
+        # GET request to Polaris Hub to list objects in path
+        response = self.polaris_client.get(ls_path.rstrip("/"), timeout=timeout)
+        response.raise_for_status()
+
+        if not detail:
+            entries = [p["name"][len(self.prefix) :] for p in response.json()]
```

Comment:
**Suggestion**
If your goal is to remove the prefix from the string, then a more reliable way is the [`removeprefix`](https://docs.python.org/3/library/stdtypes.html#str.removeprefix) method from the string class:

```suggestion
            entries = [p["name"].removeprefix(self.prefix) for p in response.json()]
```

Reply:
Learned something new here! Didn't know this existed! 🎉